src/keyboard.rs

```diff
@@ -0,0 +1,1207 @@
+impl ModifiersState {
+    /// Returns `true` if the shift key is pressed.
+    pub fn shift(&self) -> bool {
+        self.intersects(Self::SHIFT)
+    }
+    /// Returns `true` if the control key is pressed.
+    pub fn control(&self) -> bool {
+        self.intersects(Self::CONTROL)
+    }
+    /// Returns `true` if the alt key is pressed.
+    pub fn alt(&self) -> bool {
+        self.intersects(Self::ALT)
+    }
+    /// Returns `true` if the meta key is pressed.
+    pub fn meta(&self) -> bool {
+        self.intersects(Self::META)
+    }
+}
+
+bitflags! {
+    /// Represents the current state of the keyboard modifiers
+    ///
+    /// Each flag represents a modifier and is set if this modifier is active.
+    #[derive(Default)]
+    pub struct ModifiersState: u32 {
+        // left and right modifiers are currently commented out, but we should be able to support
+        // them in a future release
+        /// The "shift" key.
+        const SHIFT = 0b100 << 0;
+        // const LSHIFT = 0b010 << 0;
+        // const RSHIFT = 0b001 << 0;
+        /// The "control" key.
+        const CONTROL = 0b100 << 3;
+        // const LCTRL = 0b010 << 3;
+        // const RCTRL = 0b001 << 3;
+        /// The "alt" key.
+        const ALT = 0b100 << 6;
+        // const LALT = 0b010 << 6;
+        // const RALT = 0b001 << 6;
+        /// This is the "windows" key on PC and "command" key on Mac.
+        const META = 0b100 << 9;
```

Comment:
Calling this key "Meta" seems like a bad idea. Meta is often not clearly defined and a lot of people use the alt key as meta. The existing logo name was much more obvious, a more correct and universally understood name would probably be "Super".

Reply:
After reading a bit about this, it does seem like an awkard choice. Renaming this was intended to resolve #1343. The name which is also used in `Key` and `KeyCode` comes from the [UI Events Spec](https://w3c.github.io/uievents-key/#keys-modifier). If we want the aforementioned issue to be solved, we either have to use `Meta` or rename the corresponsing variants in `Key` and `KeyCode`. I personaly don't have a strong feeling about going either way but I do wish to use identical names throughout. `Super` seems like a good choice if we decide to rename these.

Reply:
Apparently there's this age-old keyboard called the space cadet keyboard which has both Super and Meta. I tried my best to find any kind of keyboard being sold which has both of these buttons but the ones I found are either the original space cadet for collectors or custom built keyboards. So I'd say that merging the two into just a `Super` key variant is absolutely acceptable.

Reply:
Yes, originally there was no windows key and just super/meta. Support for meta was added in many applications and super in some, so in stuff like i3/emacs/vim etc you still have references to a meta key. But since the meta key doesn't exist anymore, people remap some other key to do that. Since the super key is commonly used for more OS level shortcuts, what most people use for application meta key has become alt because of that.

So the meta key isn't the alt or the super key. But since in my experience the alt key is used by many as meta and the super key has kinda been forgotten, the latter is the perfect name for the logo key and has been used as such by a number of Linux applications I've seen.

Just as evidence that using alt as meta is indeed a very common thing, setting things up this way is even documented in the XTerm documentation: https://invisible-island.net/xterm/ctlseqs/ctlseqs.html#h2-Alt-and-Meta-Keys

Reply:
Got it!

I did rename it to super but as I was kindly reminded by the compiler, `super` is a keyword so for example in the `ModifiersState` getter functions as well as its serialized form I appended "_key" to each modifier name, e.g. `alt_key`, `super_key`. This is a bit of a compromise but I couldn't come up with a cleaner solution.